checks/branch_protection.go

```diff
@@ -84,40 +115,43 @@ func checkReleaseAndDevBranchProtection(ctx context.Context, r repositories, dl
 		}
 
 		// Try to resolve the branch name.
-		name, err := resolveBranchName(branches, *release.TargetCommitish)
+		b, err := branchesMap.getBranchByName(release.GetTargetCommitish())
 		if err != nil {
 			// If the commitish branch is still not found, fail.
 			return checker.CreateRuntimeErrorResult(CheckBranchProtection, err)
 		}
 
 		// Branch is valid, add to list of branches to check.
-		checkBranches[*name] = true
+		checkBranches[b.GetName()] = true
 	}
 
 	// Add default branch.
 	repo, _, err := r.Get(ctx, ownerStr, repoStr)
 	if err != nil {
 		return checker.CreateRuntimeErrorResult(CheckBranchProtection, err)
 	}
-	checkBranches[*repo.DefaultBranch] = true
+	checkBranches[repo.GetDefaultBranch()] = true
 
 	protected := true
 	unknown := false
 	// Check protections on all the branches.
 	for b := range checkBranches {
-		p, err := isBranchProtected(branches, b)
+		branch, err := branchesMap.getBranchByName(b)
 		if err != nil {
+			if errors.As(err, &errInternalBranchNotFound) {
+				continue
+			}
 			return checker.CreateRuntimeErrorResult(CheckBranchProtection, err)
 		}
 		// nolint
-		if !p {
+		if !branch.GetProtected() {
 			protected = false
 			dl.Warn("branch protection not enabled for branch '%s'", b)
 		} else {
 			// The branch is protected. Check the protection.
 			score, err := getProtectionAndCheck(ctx, r, dl, ownerStr, repoStr, b)
 			if err != nil {
-				if errors.Is(err, errInternalBranchNotFound) {
+				if errors.As(err, &errInternalBranchNotFound) {
```

Comment:
can you tell me the difference between `As` and `Is` and why `As` is better?

Reply:
Ah you are right! I always considered `As` being the equivalent of `check error type only and not value`. Need to update all occurring of `errors.As` in codebase.